src/main.c

```diff
@@ -117,6 +122,9 @@ static void hathor_main(void) {
     volatile unsigned int tx = 0;
     volatile unsigned int flags = 0;
 
+    // reset global state when starting
+    os_memset(&global, 0, sizeof(global));
```

Comment:
You copied this line at least three times. You should consider creating an inline method for it.

Reply:
As it's just one line, I don't see the benefit of refactoring into another method.

Reply:
I don't see why we would copy the same line over and over. If it has a bug, we have to fix in several places. Besides that, an inline method would be "the same as writing it here".

Reply:
Ok. It seems good.